fabric-levelgenerators-v1/src/main/java/net/fabricmc/fabric/mixin/levelgenerator/MixinServerPropertiesHandler.java

```diff
@@ -0,0 +1,86 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.levelgenerator;
+
+import java.util.Properties;
+import java.util.function.Function;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.spongepowered.asm.mixin.Implements;
+import org.spongepowered.asm.mixin.Interface;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import net.minecraft.server.dedicated.AbstractPropertiesHandler;
+import net.minecraft.server.dedicated.ServerPropertiesHandler;
+import net.minecraft.util.Identifier;
+import net.minecraft.world.level.LevelGeneratorType;
+
+import net.fabricmc.fabric.impl.levelgenerator.ServerPropertiesHandlerImplements;
+
+@Mixin(ServerPropertiesHandler.class)
+@Implements(@Interface(iface = ServerPropertiesHandlerImplements.class, prefix = "fabric$"))
+public abstract class MixinServerPropertiesHandler extends AbstractPropertiesHandler<ServerPropertiesHandler> implements ServerPropertiesHandlerImplements {
+	private static final Logger LOGGER = LogManager.getLogger();
+	private Identifier fabriclevelType;
+
+	public MixinServerPropertiesHandler(Properties properties) {
+		super(properties);
+	}
+
+	@SuppressWarnings("unchecked")
+	@Redirect(method = "<init>", at = @At(value = "INVOKE", target = "Lnet/minecraft/server/dedicated/ServerPropertiesHandler;get(Ljava/lang/String;Ljava/util/function/Function;Ljava/util/function/Function;Ljava/lang/Object;)Ljava/lang/Object;", ordinal = 2))
+	private <V> V replaceLevelType(ServerPropertiesHandler serverPropertiesHandler, String prop, Function<String, V> function, Function<V, String> function2, V defaultObject) {
+		if (!prop.equals("level-type")) throw new RuntimeException("This mixin should only be applied to level-type");
+
+		// If level-type doesn't exist, fallback to default behavior
+		if (getProperties().get(prop) == null) {
+			get("level-type", LevelGeneratorType::getTypeFromName, LevelGeneratorType::getName, LevelGeneratorType.DEFAULT);
+		}
+
+		// Parse String as segments
+		String value = getProperties().get(prop).toString();
+		String[] levelType = value.split(":");
+
+		// Check for vanilla levelGenerators
+		LevelGeneratorType levelGeneratorType = LevelGeneratorType.getTypeFromName(value);
+		if (levelGeneratorType != null) return (V) levelGeneratorType;
+
+		// Gives ability to skip namespace
+		if (levelType.length == 1) {
+			fabriclevelType = new Identifier("fabricdefault", levelType[0]);
```

Comment:
Perhaps a namespace that is more explicit that this is a default.
```suggestion
			fabriclevelType = new Identifier("fabric_omitted_namespace", levelType[0]);
```